Architecture.md

```diff
@@ -73,10 +95,49 @@
 
 #### Client Applications
 
-***TODO: [INFRA] how are the Client applications provisioned, how they are exposed and how the underlying infrastructure is managed***
+Pomegranate is responsible for starting, stopping, and interacting with the client applications.
+It is also responsible for managing the networking of the applications.
+
+Pomegranate is completely stateless, and the API is the only external way to interact with it.
+Once a container is started, Pomegranate does not keep track of it, and it is the responsibility of the execution engine to manage it.
+
+```mermaid
+flowchart TD
+    web[80]
+    webscure[443]
+    traefik[Traefik]
+    app_1[User app 1]
+    app_2[User app 2]
+    app_3[User app 3]
+
+    rules{Resolve\nvia labels}
+
+    webscure --> |HTTPS| traefik
+    web --> |HTTP| traefik
+    traefik --> rules
+
+    rules --> |uuid_1.user-app:80| app_1
+    rules --> |uuid_2.user-app:80| app_2
+    rules --> |uuid_3.user-app:80| app_3
```

Comment:
Suggest adding bold to UUIDs to make them stand out, and add the FQDN to the end of the URL

```suggestion
    rules --> |"`**uuid1**.user-app.**FQDN**:80`"| app_1
    rules --> |"`**uuid2**.user-app.**FQDN**:80`"| app_2
    rules --> |"`**uuid3**.user-app.**FQDN**:80`"| app_3
```

Reply:
While the suggestion makes sense for the uuid, the FQDN part is just overloading the graph for very little added value.

Reply:
We can find a compromise in that, however I would still suggest defining FQDN elsewhere (maybe after the graph) in order for the reader to follow along.